### PR TITLE
[COTCO-847] Region summary page implemented

### PR DESCRIPTION
**Pull request by Flexicoder**

### CODE REVIEW

File: app/uk/gov/hmrc/ootcompletion/connectors/RegionConnector.scala [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+import uk.gov.hmrc.ootcompletion.models.api._
+import uk.gov.hmrc.http.HeaderCarrier
+import uk.gov.hmrc.play.bootstrap.http.HttpClient
+
+@Singleton
+class RegionConnector @Inject()(appConfig: AppConfig, httpClient: HttpClient)(implicit ec: ExecutionContext) {
+  private val url = appConfig.ootcBaseUrl
+
+  def getRegionPortViewData(region: String)(implicit headerCarrier: HeaderCarrier): Future[Region] =
+    httpClient.GET[Region](s"$url/aggregates/region/$region")
```
Comment (peter-hazell) [inline app/uk/gov/hmrc/ootcompletion/connectors/RegionConnector.scala]: The endpoint will be:

/pending-transits-summary/region/:region

File: app/uk/gov/hmrc/ootcompletion/connectors/RegionConnector.scala [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+import uk.gov.hmrc.ootcompletion.config.AppConfig
+import uk.gov.hmrc.ootcompletion.models.api._
+import uk.gov.hmrc.http.HeaderCarrier
+import uk.gov.hmrc.play.bootstrap.http.HttpClient
+
+@Singleton
+class RegionConnector @Inject()(appConfig: AppConfig, httpClient: HttpClient)(implicit ec: ExecutionContext) {
+  private val url = appConfig.ootcBaseUrl
+
+  def getRegionPortViewData(region: String)(implicit headerCarrier: HeaderCarrier): Future[Region] =
```
Comment (peter-hazell) [inline app/uk/gov/hmrc/ootcompletion/connectors/RegionConnector.scala]: can we name this something like 'getPendingTransitsRegionSummary'

File: app/uk/gov/hmrc/ootcompletion/controllers/PendingMRNSummaryController.scala [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+import play.api.{Configuration, Environment}
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
+
+import scala.concurrent.Future
+import uk.gov.hmrc.ootcompletion.config.AppConfig
+import uk.gov.hmrc.ootcompletion.connectors.StrideAuthConnector
+import uk.gov.hmrc.ootcompletion.services.RegionService
+
+@Singleton
+class PendingMRNSummaryController @Inject()(
```
Comment (peter-hazell) [inline app/uk/gov/hmrc/ootcompletion/controllers/PendingMRNSummaryController.scala]: Can we call this PendingTransitsSummaryController

File: app/uk/gov/hmrc/ootcompletion/connectors/RegionConnector.scala [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+import scala.concurrent.{ExecutionContext, Future}
+
+import com.google.inject.Singleton
+import uk.gov.hmrc.ootcompletion.config.AppConfig
+import uk.gov.hmrc.ootcompletion.models.api._
+import uk.gov.hmrc.http.HeaderCarrier
+import uk.gov.hmrc.play.bootstrap.http.HttpClient
+
+@Singleton
+class RegionConnector @Inject()(appConfig: AppConfig, httpClient: HttpClient)(implicit ec: ExecutionContext) {
```
Comment (peter-hazell) [inline app/uk/gov/hmrc/ootcompletion/connectors/RegionConnector.scala]: Generally you'd have one connector per microservice you need to call. I think it'd be cleaner if we changed the frontend to do this. Not something for this PR though

File: conf/app.routes [hunk 5]
```diff
@@ -9,6 +9,10 @@ POST        /manager                                                @uk.gov.hmrc
 
 GET         /manager/region/:selectedRegion                         @uk.gov.hmrc.ootcompletion.controllers.RegionPortSelectionController.showRegionPortSelectionPage(selectedRegion: String)
 
+GET         /manager/region/:selectedRegion/Pending/:selectedPort   @uk.gov.hmrc.ootcompletion.controllers.PendingMRNSummaryController.showSummaryPage(selectedRegion: String, selectedPort: String)
```
Comment (peter-hazell) [inline conf/app.routes]: Pending and Rejected should all be lower case I think

File: app/uk/gov/hmrc/ootcompletion/models/api/AggregationTables.scala [hunk 6]
```diff
@@ -38,3 +40,11 @@ object Estimation {
 object DestinationPort {
   implicit val format: Format[DestinationPort] = Json.format[DestinationPort]
 }
+
+object Region {
```
Comment (peter-hazell) [inline app/uk/gov/hmrc/ootcompletion/models/api/AggregationTables.scala]: Can we move these models into a different file? As AK is doing some work to move away from these aggregation table models.

Also in the backend i've called the models 

`case class PendingTransitsRegionSummary(name: String, ports: List[PendingRegionPort])`

and 

`case class PendingRegionPort(name: DestinationPort, pendingMrnCount: Int, rejectedMrnCount: Int)`

File: app/uk/gov/hmrc/ootcompletion/views/region_port_selection.scala.html [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+
+@main_template(
+    title = Messages("region_port_selection.title"),
+    bodyClasses = None
+) {
+    <article class="full-width">
+        <a href="@RegionSelectionController.showRegionSelectionPage()" class="back-link">Back</a>
+        <h1 class="heading-xlarge ctc-margin-bottom-3">@region.name</h1>
+            @for(regionPort <- region.ports) {
+            <div class="ctc-margin-top-2 ">
```
Comment (peter-hazell) [inline app/uk/gov/hmrc/ootcompletion/views/region_port_selection.scala.html]: Just wondering if we absolutely need to use this custom css ? are there not some styles out of the box that give us what we need?

File: app/uk/gov/hmrc/ootcompletion/views/region_port_selection.scala.html [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+
+@import uk.gov.hmrc.ootcompletion.models.api.Region
+
+@(region: Region)(implicit request: AuthRequest[_], messages: Messages, appConfig: AppConfig)
+
+@main_template(
+    title = Messages("region_port_selection.title"),
+    bodyClasses = None
+) {
+    <article class="full-width">
```
Comment (peter-hazell) [inline app/uk/gov/hmrc/ootcompletion/views/region_port_selection.scala.html]: Do we need article ?

File: app/uk/gov/hmrc/ootcompletion/connectors/RegionConnector.scala (hunk 2 above)
Comment (Flexicoder) [reply app/uk/gov/hmrc/ootcompletion/connectors/RegionConnector.scala]: I've made this change, but just wanted to double check, this end point will return both a count for pending and rejected transits

Comment (peter-hazell) [reply app/uk/gov/hmrc/ootcompletion/connectors/RegionConnector.scala]: Ahhh yes - good point, in which case, let's take the word pending out and go with `getTransitsRegionSummary`

File: app/uk/gov/hmrc/ootcompletion/connectors/RegionConnector.scala (hunk 1 above)
Comment (peter-hazell) [reply app/uk/gov/hmrc/ootcompletion/connectors/RegionConnector.scala]: Updating this endpoint to be: `/transits-summary/region/:region` as per other comment

File: app/uk/gov/hmrc/ootcompletion/models/api/AggregationTables.scala (hunk 6 above)
Comment (Flexicoder) [reply app/uk/gov/hmrc/ootcompletion/models/api/AggregationTables.scala]: Should these be renamed as well?

Comment (peter-hazell) [reply app/uk/gov/hmrc/ootcompletion/models/api/AggregationTables.scala]: yup - will take the word Pending out

Comment (peter-hazell) [reply app/uk/gov/hmrc/ootcompletion/models/api/AggregationTables.scala]: So I've gone with:

`case class TransitsRegionSummary(name: String, ports: List[TransitsRegionPort])`

`case class TransitsRegionPort(name: DestinationPort, pendingMrnCount: Int, rejectedMrnCount: Int)`

File: app/uk/gov/hmrc/ootcompletion/views/region_port_selection.scala.html (hunk 7 above)
Comment (Flexicoder) [reply app/uk/gov/hmrc/ootcompletion/views/region_port_selection.scala.html]: I've removed the article element and the margin top, I've had to use the existing `ctc-margin-bottom-4` to get padding between the port headings, I've tried the standard css classes that are meant to exist but they aren't working

File: app/uk/gov/hmrc/ootcompletion/models/api/AggregationTables.scala (hunk 6 above)
Comment (Flexicoder) [reply app/uk/gov/hmrc/ootcompletion/models/api/AggregationTables.scala]: @peter-hazell is that first parameter name `name` correct - `case class TransitsRegionPort(name: DestinationPort, pendingMrnCount: Int, rejectedMrnCount: Int)` so to get the actual name I need to do `name.name`? Shouldn't it be called `port` or something?

File: app/uk/gov/hmrc/ootcompletion/controllers/RejectedMRNSummaryController.scala [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+import play.api.{Configuration, Environment}
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
+
+import scala.concurrent.Future
+import uk.gov.hmrc.ootcompletion.config.AppConfig
+import uk.gov.hmrc.ootcompletion.connectors.StrideAuthConnector
+import uk.gov.hmrc.ootcompletion.services.RegionService
+
+@Singleton
+class RejectedMRNSummaryController @Inject()(
```
Comment (peter-hazell) [inline app/uk/gov/hmrc/ootcompletion/controllers/RejectedMRNSummaryController.scala]: Can we call this RejectedTransitSummaryController

Comment (Flexicoder) [reply app/uk/gov/hmrc/ootcompletion/controllers/RejectedMRNSummaryController.scala]: yep already done :)

File: app/uk/gov/hmrc/ootcompletion/models/api/AggregationTables.scala (hunk 6 above)
Comment (peter-hazell) [reply app/uk/gov/hmrc/ootcompletion/models/api/AggregationTables.scala]: You would get the region name from TransitRegionSummary and the port name from the list of ports, so something like `ports.map(_.name)`

However, for clarity and to align with how things are named in the backend , I've done some further renaming:

`case class TransitsRegionSummary(region: Region, ports: List[TransitsRegionPort])`

`case class TransitsRegionPort(destinationPort: DestinationPort, pendingMrnCount: Int, rejectedMrnCount: Int)` 

I've added a Region type to the refined types as well

**Comment by peter-hazell** [comment]
A further comment on this PR. 

Can we get rid of the `GET         /reports/manager` route now which routes to the National Hub page.

Instead have the logic in the `/manager/region/:selectedRegion` route to do a redirect to the National Hub page if the user has selected National hub 

This will make it more consistency with a single route through to all the relevant places.

**Comment by peter-hazell** [comment]
@Flexicoder - The backend work for this is now complete and merged, so in your absence, I have pushed a few updates to this branch. 

- Aligned models with the backend API models
- Some minor naming updates/refactoring
- Updated our scalafmt conf and ran scalafmt across the whole codebase
- Updated the URL for the Hub rejected page as per Marcus' comment on COTCO-818
- Rebased with master

From my POV, tests are passing and I have tested locally and FE + BE are hooking up correctly, so this is in a mergeable state now. Happy to run through Backend with you and latest updates.

See you tomorrow!

**Pull request merged**